src/test/typescriptLambdaHandlerSearch.test.ts

```diff
@@ -0,0 +1,123 @@
+                'sampleFunctions.exportedFunctionWithNoArgs',
+                'sampleFunctions.exportedFunctionWithOneArg',
+                'sampleFunctions.exportedFunctionWithTwoArgs',
+                'sampleFunctions.exportedFunctionWithThreeArgs'
+            ])
+
+            testTypescriptLambdaHandlerSearch(filename, expectedHandlerNames)
+        })
+
+        it('ignores class functions', async () => {
+            const filename: string = path.join(
+                getSamplesFolder(),
+                'typescript',
+                'sampleClasses.ts')
+
+            const expectedHandlerNames: Set<string> = new Set([
+                'sampleClasses.exportedFunctionWithNoArgs',
+            ])
+
+            testTypescriptLambdaHandlerSearch(filename, expectedHandlerNames)
+        })
+
+        it('ignores interface functions', async () => {
+            const filename: string = path.join(
+                getSamplesFolder(),
+                'typescript',
+                'sampleInterfaces.ts')
+
+            const expectedHandlerNames: Set<string> = new Set([
+                'sampleInterfaces.exportedFunctionWithNoArgs',
+            ])
+
+            testTypescriptLambdaHandlerSearch(filename, expectedHandlerNames)
+        })
+
+    })
+
+    describe('Javascript code', () => {
+
+        it('finds exported functions', async () => {
+            const filename: string = path.join(
+                getSamplesFolder(),
+                'javascript',
+                'sampleFunctions.js')
+
+            const expectedHandlerNames: Set<string> = new Set([
+                'sampleFunctions.exportedFunctionWithNoArgs',
+                'sampleFunctions.exportedFunctionWithOneArg',
+                'sampleFunctions.exportedFunctionWithTwoArgs',
+                'sampleFunctions.exportedFunctionWithThreeArgs'
+            ])
+
+            testTypescriptLambdaHandlerSearch(filename, expectedHandlerNames)
+        })
+
+        it('ignores class functions', async () => {
+            const filename: string = path.join(
+                getSamplesFolder(),
+                'javascript',
+                'sampleClasses.js')
+
+            const expectedHandlerNames: Set<string> = new Set([
+                'sampleClasses.exportedFunctionWithNoArgs',
+            ])
+
+            testTypescriptLambdaHandlerSearch(filename, expectedHandlerNames)
+        })
+
+    })
+
+    async function testTypescriptLambdaHandlerSearch(
+        filename: string,
+        expectedHandlerNames: Set<string>
+    ): Promise<void> {
+        const search: TypescriptLambdaHandlerSearch = new TypescriptLambdaHandlerSearch(
+            vscode.Uri.file(filename)
+        )
+
+        const handlers: LambdaHandlerCandidate[] = await search.findCandidateLambdaHandlers()
+
+        assertCandidateHandlers(handlers, expectedHandlerNames)
+    }
+
+    function getSamplesFolder(): string {
+        return path.join(
```

Comment:
Why not just `./samples`?

Reply:
Typescript compiles the sample ts files to js under the /out folder, so I am referencing the source files.

I didn't want to cut the extension off the files, so that they were easier to author